src/main/java/it/gov/pagopa/payhub/activities/service/debtposition/ionotification/DebtOperationOperationTypeResolver.java

```diff
@@ -0,0 +1,64 @@
+package it.gov.pagopa.payhub.activities.service.debtposition.ionotification;
+
+import it.gov.pagopa.pu.debtposition.dto.generated.DebtPositionDTO;
+import it.gov.pagopa.pu.debtposition.dto.generated.InstallmentDTO;
+import it.gov.pagopa.pu.debtposition.dto.generated.IupdSyncStatusUpdateDTO;
+import it.gov.pagopa.pu.ionotification.dto.generated.NotificationRequestDTO;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Lazy
+@Component
+public class DebtOperationOperationTypeResolver {
+
+    private final InstallmentOperationTypeResolver installmentOperationTypeResolver;
+
+    public DebtOperationOperationTypeResolver(InstallmentOperationTypeResolver installmentOperationTypeResolver) {
+        this.installmentOperationTypeResolver = installmentOperationTypeResolver;
+    }
+
+    public NotificationRequestDTO.OperationTypeEnum calculateDebtPositionOperationType(DebtPositionDTO debtPositionDTO, Map<String, IupdSyncStatusUpdateDTO> iupdSyncStatusUpdateDTOMap) {
+
```

Comment:
if iupdSyncStatusUpdateDTOMap.isEmpty you should return null -> no op performed, thus not notification

Reply:
the check on iupdSyncStatusUpdateDTOMap is to be performed as first!
it should be executed before to check if the DP has active Installments